editor/scene_tree_dock.cpp

```diff
@@ -2224,17 +2224,111 @@ void SceneTreeDock::_toggle_placeholder_from_selection() {
 	}
 }
 
+void SceneTreeDock::_update_all_gizmos(Node *p_node) {
+	Node3DEditor::get_singleton()->update_all_gizmos(p_node);
+}
+
+void SceneTreeDock::_reparent_nodes_to_root(Node *p_root, Array p_nodes, Node *p_owner) {
+	List<Node *> nodes;
+	for (int i = 0; i < p_nodes.size(); i++) {
+		Node *node = Object::cast_to<Node>(p_nodes[i]);
+		ERR_FAIL_COND(node == nullptr);
+		nodes.push_back(node);
+	}
+
+	for (Node *node : nodes) {
+		node->set_owner(p_owner);
```

Comment:
So I'm a little confused what you mean. The if statement is basically meant to check if the nodes in the array are parented to nodes owned by the instance we just removed editable children status from.

Admitedly though, after looking over it, I think this was the wrong place to put it. Instead, I've put it up in _toggle_editable_children when constructing the initial arrays, because otherwise, there's discontinuity between the two methods and this check doesn't reflect the description of what this specific method is meant to do.

Reply:
What I meant was that `reparent()` clears the owner, so setting it before reparenting is pointless (you are still doing it). 

Previously your reparenting was conditional, so the unconditional `set_owner()` should go under else of that condition. This part is no longer relevant.